### ISSUE TITLE
Why is the training loop an infinite loop?

### ISSUE THREAD

**Issue by bonnie-love-ai** [llSourcell/tensorflow_speech_recognition_demo#12]
Maybe this question is silly, but I do not get why the training loop is "while 1". Why wouldn't that train  for an infinite number of times?

**Comment by MatthewScholefield** [comment]
I'm guessing this should be replaced by something like `for _ in range(training_iters)`

**Comment by arvind1705** [comment]
Because number of steps is set to 30000, unfortunately this takes a lot of time.So this seems infinite.
Reduce the number iterations.

**Comment by MatthewScholefield** [comment]
@arvind1705 No, in the demo code it literally is an infinite loop. The `training_iters` is just never used.

**Comment by djpg** [comment]
You have to replace `while 1: #training_iters:` to `for i in range(training_iters):`
